### PR TITLE
Student immatriculation

### PR DESCRIPTION
**Pull request by bastihav**
# Description
- Closes #749 
Implements student immatriculation using frontend signing

## Type of change (remove all that don't apply)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [X] Manually tested all use cases this could alter

**Test Configuration**:
- OS: Windows Mac Linux
- Browser: Firefox Chrome Safari Chromium

- Frontend: (remove all that don't apply)
  - [X] Development build
- Backend: (remove all that don't apply)
  - [ ] deployed production build
  - [ ] deployed development build
  - [X] deployed experimental build

# Checklist: (remove all that don't apply)

- [x] I added corresponding E2E tests (especially for bugfixes)
- [x] I added corresponding unit tests (if API code)
- [X] I have performed a self-review of my own code
- [X] My changes generate no new linting warnings or console warnings
- [x] I have updated the CHANGELOG.md to include any changes made in this PR (add WIP to the top, if there is none already)

### CODE REVIEW

**Review by tobias-wiese — APPROVED** [review]
We may consider moving the code for getting the unsigned proposal, keys etc. to the `MatriculationManagement` s.t. the component just calls something like `await matriculationManagement.updateMatriculation(data)` and all the intermediate steps are abstacted from the component.

**Comment by bastihav** [comment]
> We may consider moving the code for getting the unsigned proposal, keys etc. to the `MatriculationManagement` s.t. the component just calls something like `await matriculationManagement.updateMatriculation(data)` and all the intermediate steps are abstacted from the component.

I added a new abstraction folder in the api, which will handle those kinds of abstractions in the future. Please re-review, considering that a large portion of the code was refactored and updated due to the new transaction API changes.

**Review by tobias-wiese — APPROVED** [review]

File: src/use/store/getters.ts [hunk 1]
```diff
@@ -59,6 +61,25 @@ export const getters: GetterTree<State, State> & Getters = {
                 }
 
                 store.commit(MutationTypes.SET_PRIVATE_KEY, privateKey);
+            } else {
+                // no key at lagom?
```
Comment (theovier) [inline src/use/store/getters.ts:65]: ?

**Review by theovier — APPROVED** [review]

File: src/use/crypto/signing.ts [hunk 2]
```diff
@@ -121,6 +122,31 @@ export async function verifyProposalSignature(proposal: string, signature: strin
...
+
+    // const berObject = (asn1js.fromBER(base64ToArrayBuffer(signature)).result.toJSON() as any).valueBlock.value;
+
+    // const r = fromHexString(berObject[0].valueBlock.valueHex);
+    // const s = fromHexString(berObject[1].valueBlock.valueHex);
+
+    // const sig = { r, s };
+
+    // return _checkMalleability(sig, { name: signingAlgorithm.namedCurve }) && p256ec.verify(new Uint8Array(digest), sig, exportedPublicKey);
+
```
[the remark this answers is not in the document]
Comment (bastihav) [reply src/use/crypto/signing.ts:146]: I think we should keep this comment as it is important to document that we don't actually check anything here and why.

I do not think we can just put that in an issue and forget, if someone looks at the code, they will not look through all issues to find a comment on why we do not verify signatures yet.

File: src/use/store/getters.ts (hunk 1 above)
Comment (bastihav) [reply src/use/store/getters.ts]: I updated the comment.

**Comment by tobias-wiese** [comment]
I adapted the E2E tests again. They could not succeed because we still had the matriculationManagement tests for the admin. I moved some checks to the new studentImmatriculation test file and deleted the old one. Please have a look into that @bastihav

**Pull request merged**